rust/services/workload/src/types.rs

```diff
@@ -6,8 +6,10 @@ use std::collections::HashMap;
 use strum_macros::AsRefStr;
 
 #[derive(Serialize, Deserialize, Clone, Debug)]
-pub struct ObjectIdJSON {
+pub struct HostIdJSON {
     pub _id: ObjectId,
+    #[serde(skip_serializing_if = "String::is_empty")]
+    pub device_id: String,
```

Comment:
i'd like us to commit to one semantic here:

Reply:
1. optional

```suggestion
    #[serde(skip_serializing_if = "Option::is_none")]
    pub maybe_device_id: Option<String>,
```

Reply:
2. required

1. optional

```suggestion
    pub device_id: String,
```

Reply:
Let's go with 2.